evm/src/main/java/org/hyperledger/besu/evm/fluent/SimpleWorld.java

```diff
@@ -120,12 +123,16 @@ public void revert() {
 
   @Override
   public void commit() {
-    accounts.forEach(
-        (address, account) -> {
-          if (!account.updateParent()) {
-            parent.accounts.put(address, account);
-          }
-        });
+    accounts.entrySet().stream()
+        .filter(e -> e.getValue().isPresent())
```

Comment:
Shouldn't a commit to the parent also commit all deletes?  Doesn't this filter make all deletes transient?  EIP-6780 compatible, but that should be advertised in the docs that SimpleWorld is not suitable for pre-shanghai EVMs.

Reply:
Well spotted @shemnon I would argue that it wasn't committing deletes before as it would throw an NPE, so I was unsure of the right behavior. But now that you mention it, not committing doesn't make sense at all - I'll change.